compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -526,10 +526,7 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
                     // anyway, except via auto trait matching (which
                     // only inspects the upvar types).
                     walker.skip_current_subtree(); // subtree handled below
-                    for upvar_ty in substs.as_closure().upvar_tys() {
-                        // FIXME(eddyb) add the type to `walker` instead of recursing.
-                        self.compute(upvar_ty.into());
-                    }
+                    self.compute(substs.as_closure().tupled_upvars_ty().into());
```

Comment:
It feels like at this point we would've done capture analysis, based on the comments, but not sure.

Also how `ty::Tuple` is dealt with within this code seems to be different than what was done here before this change.